adapter/operations.go

```diff
@@ -16,13 +16,21 @@ package adapter
 
 import "context"
 
-const (
-	OperationDescriptionKey  = "description"
-	OperationVersionKey      = "version"
-	OperationTemplateNameKey = "templateName"
-	OperationServiceNameKey  = "serviceName"
+var (
+	NoneVersion = []Version{"none"}
 )
 
+type Version string
+
+type Operation struct {
+	Type        int32     `json:"type,string,omitempty"`
+	Description string    `json:"description,omitempty"`
+	Versions    []Version `json:"versions,omitempty"`
+	Template    string    `json:"template,omitempty"`
```

Comment:
Not all operations are using a template, e.g. installing osm. Should we have this as a fixed field?

Reply:
Yes @mgfeller  So that gets omitted if its not in use, just introduces some flexibility